src/main/kotlin/com/google/androidstudiopoet/models/Topologies.kt

```diff
@@ -26,6 +27,7 @@ import java.util.*
 enum class Topologies {
     FULL {
         override fun generateDependencies(parameters: Map<String, String>, configPOJO: ConfigPOJO): List<FromToDependencyConfig> {
+            val method = getMethod(parameters)
```

Comment:
Not for this PR, but we probably need to  find a way to refactor these topologies a bit, because they look very similar. Although I'm not sure if that is possible.

Reply:
Every time I modify this file I think the same, there should be a way to have all common initialization in a single place. I think it can be done by moving all common variables to be private members and have a method that initialize them. The structure of generateDependencies will be like this:

- Call common initialization
- Initialize specific parameters
- Check for non used parameters, to prevent typo errors from input (this is not done yet)
- Generate dependencies

Once this PR is merged I will work on having that.

Reply:
It would be awesome if we can bring topologies to the Full format. But that'll require some work to be done :)

Reply:
PR #107 implements the ideas from my previous comment.

I also want to make them work with the Full format, right now they depend on the names of the modules, that can be easily solved by passing this list instead of having it hard coded as it is now. I can do this as a first step to have them in the Full format. One benefit of having topologies of doing that is that it will be possible to apply topologies only to subsets of modules.